mempool/src/shared_mempool/tasks.rs

```diff
@@ -238,6 +238,16 @@ pub(crate) async fn process_transaction_broadcast<V>(
 {
     let results = process_incoming_transactions(&smp, transactions, timeline_state).await;
     log_txn_process_results(&results, Some(peer.peer_id()));
+
+    // security log
+    for (_mempool_status, maybe_vm_status) in &results {
+        if let Some((transaction, _vm_status)) = maybe_vm_status {
+            send_struct_log!(security_log(security_events::INVALID_TRANSACTION_MP)
+                .data("transaction", &transaction)
+                .data("from_peer", &peer));
```

Comment:
logging the transaction and the peer who sent it to us

Reply:
why don't we just keep track of the transactions and then zip the results together? It seems unfortunate to require that we return the transaction when we already have it... and then we have to modify a bunch of other APIs too.

Reply:
I did start doing this, but two things that went in my head when I made this change that made me go this way:

- I wanted to avoid using functions that can crash if the length, for any reason, ends up differing (after a refactor for example).
- I thought it would be useful to return the transaction along with a failure

That being said, it is only used in one place for now. Wdyt?